src/jit/compiler.h

```diff
@@ -605,61 +612,15 @@ class LclVarDsc
     unsigned m_lvRefCntWtd;    // weighted reference count
 
 public:
-    unsigned short lvRefCnt() const
-    {
-        if (lvImplicitlyReferenced && (m_lvRefCnt == 0))
-        {
-            return 1;
-        }
-
-        return m_lvRefCnt;
-    }
-
-    void incLvRefCnt(unsigned short delta)
-    {
-        unsigned short oldRefCnt = m_lvRefCnt;
-        m_lvRefCnt += delta;
-        assert(m_lvRefCnt >= oldRefCnt);
-    }
+    unsigned short lvRefCnt(RefCountState state = RCS_NORMAL) const;
+    void incLvRefCnt(unsigned short delta, RefCountState state = RCS_NORMAL);
+    void decLvRefCnt(unsigned short delta, RefCountState state = RCS_NORMAL);
+    void setLvRefCnt(unsigned short newValue, RefCountState state = RCS_NORMAL);
 
-    void decLvRefCnt(unsigned short delta)
-    {
-        assert(m_lvRefCnt >= delta);
-        m_lvRefCnt -= delta;
-    }
-
-    void setLvRefCnt(unsigned short newValue)
-    {
-        m_lvRefCnt = newValue;
-    }
-
-    unsigned lvRefCntWtd() const
-    {
-        if (lvImplicitlyReferenced && (m_lvRefCntWtd == 0))
-        {
-            return BB_UNITY_WEIGHT;
-        }
-
-        return m_lvRefCntWtd;
-    }
-
-    void incLvRefCntWtd(unsigned delta)
-    {
-        unsigned oldRefCntWtd = m_lvRefCntWtd;
-        m_lvRefCntWtd += delta;
-        assert(m_lvRefCntWtd >= oldRefCntWtd);
-    }
-
-    void decLvRefCntWtd(unsigned delta)
-    {
-        assert(m_lvRefCntWtd >= delta);
-        m_lvRefCntWtd -= delta;
-    }
-
-    void setLvRefCntWtd(unsigned newValue)
-    {
-        m_lvRefCntWtd = newValue;
-    }
+    unsigned lvRefCntWtd(RefCountState state = RCS_NORMAL) const;
+    void incLvRefCntWtd(unsigned delta, RefCountState state = RCS_NORMAL);
+    void decLvRefCntWtd(unsigned delta, RefCountState state = RCS_NORMAL);
+    void setLvRefCntWtd(unsigned newValue, RefCountState state = RCS_NORMAL);
```

Comment:
We probably want the weighted ref counts to eventually be 32-bit floats.
And BasicBlocks already use a user type `weight_t`.
We should probabaly switch the type used for weighted counts over to `weight_t`

Reply:
Ok, will update.